src/global/css.ts

```diff
@@ -11,50 +11,35 @@ let isDirty = false;
 
 // These only make sense on the browser. See additional setup below.
 let styleElement = null;
-let useCssText = null;
 
 // flag to use multiple style tags
 let _splitTag = false;
-export function splitTag(v) {
+export function splitTag(v: boolean): boolean {
   return v === undefined ? _splitTag : (_splitTag = v);
 }
 
-export function addCSS(styleDefinition) {
-  styleDefinitions.push(styleDefinition);
-  isDirty = true;
-}
+function makeStyle(id?: string): HTMLStyleElement {
+  if (doc) {
+    const el = doc.createElement('style');
+    el.type = 'text/css';
+    el.setAttribute('data-ractive-css', id || '');
 
-export function applyCSS(force) {
-  const styleElement = style();
+    doc.getElementsByTagName('head')[0].appendChild(el);
+    return el;
+  }
+}
 
-  // Apply only seems to make sense when we're in the DOM. Server-side renders
-  // can call toCSS to get the updated CSS.
-  if (!styleElement || (!force && !isDirty)) return;
+function style(): HTMLStyleElement {
+  if (!styleElement) styleElement = makeStyle();
 
-  if (_splitTag) {
-    styleDefinitions.forEach(s => {
-      const el = getStyle(s.id);
-      if (el) {
-        const css = getCSS(s.id);
-        if (useCssText) {
-          el.styleSheet.cssText !== css && (el.styleSheet.cssText = css);
-        } else {
-          el.innerHTML !== css && (el.innerHTML = css);
-        }
-      }
-    });
-  } else {
-    if (useCssText) {
-      styleElement.styleSheet.cssText = getCSS(null);
-    } else {
-      styleElement.innerHTML = getCSS(null);
-    }
```

Comment:
Hi @evs-chris,
I have removed the use of `useCssText` variable because it works with `styleSheet` property which is not available on `HTMLStyleElement`. Does this makes sense to you?

Reply:
It does, but it brings up a question of legacy browser support. This was only here to support IE with its funky non-standard properties. Personally, I'm for dropping support for IE and legacy edge, but ractive has supported them for forever.

Reply:
I think IE11 should be supported

Reply:
I see that Ractive drop support for IE8 from version 0.8.0. So I tested the current build on IE 9, 10 and 11 (using IE 11 simulator) and with `innerHTML` it works fine.

(Click on image to view it full size)

### IE 9
<img alt="ractive-css-ie9" src="https://user-images.githubusercontent.com/24919330/82125330-939a9180-97a5-11ea-96e9-91e4939425fa.png" width="150">

### IE 10
<img alt="ractive-css-ie10" src="https://user-images.githubusercontent.com/24919330/82125331-95645500-97a5-11ea-8fe7-db7c8df07f76.png" width="150">

### IE 11
<img alt="ractive-css-ie11" src="https://user-images.githubusercontent.com/24919330/82125332-96958200-97a5-11ea-9e7e-7c9b53148e94.png" width="150">

Anyway if you feel more comfortable with the previous implementation just let me know and I will restore it.

Reply:
Most of the weird stuff has comments around it, usually pointing at IE or Safari, so as long as it doesn't break a supported browser, I'm fine with it. I'm also fine with dropping IE support, as we could term the ts-ified version ractive 2.0 and I can continue backporting bugfixes for the last 1.x branch on-demand. This particular weird thing may have been for an older Safari, which I've never had access to to test, but either way, I'm fine with it, as html5 is good enough for legacy now.

Reply:
> Most of the weird stuff has comments around it, usually pointing at IE or Safari, so as long as it doesn't break a supported browser, I'm fine with it.

I totally agree with that.

> I'm also fine with dropping IE support, as we could term the ts-ified version ractive 2.0 and I can continue backporting bugfixes for the last 1.x branch on-demand.

Maybe some people who use Ractive have to support IE11 and may not update to version 2 if it doesn't support IE11, so I agree with @kouts.

> This particular weird thing may have been for an older Safari, which I've never had access to to test, but either way, I'm fine with it, as html5 is good enough for legacy now.

🥳

Reply:
I would suggest keeping support for IE11, to make It easier for people to keep upgrading Ractive from earlier versions. I'm not against introducing breaking changes, but removing browser support is tricky, since I'm sure a lot of people using Ractive are still forced to support IE11.

I'm one of them 🙌 😭